### ISSUE TITLE
softmax axtivation

### ISSUE THREAD

**Issue by DennisRutjes** [denizyuret/Knet.jl#300]
Thank you for this great Library!

I have a neural network which has a 12x50 result array, I want a softmax on the individual rows(12)

function softmax(xs)
     exp.(xs) ./ sum(exp.(xs))
end

but  
grads = lossgrad(x,y)

ERROR: DimensionMismatch("arrays could not be broadcast to a common size")

How can this be achieved? seems softmax is not present anymore in the Knet lib.

with kind regards,

Dennis

**Comment by ozanarkancan** [comment]
Hi,

You can use the [nll](http://denizyuret.github.io/Knet.jl/latest/reference.html#Knet.nll) function.

**Comment by Evizero** [comment]
You'll want to use `nll` as loss, which works on the logits directly. logits are also sufficient to predict the class using `argmax`.

When you do want to produce probabilities you can use `exp.(logp(...))`, which is softmax. Note that `logp` has a `dim` keyword. Check out the docs for more info.

**Comment by DennisRutjes** [comment]
thx!, will give that a try
